File: cdp-langchain/examples/wallet/wallet.py
import json
import sys

# Import CDP Agentkit Langchain Extension.
from cdp_langchain.utils import CdpAgentkitWrapper


def initialize_agentkit():
    """Initialize the CDP Agentkit

    Returns:
        tuple: Agent executor and config

    """
    config = {}
    agentkit = CdpAgentkitWrapper(**config)

    return agentkit, config


def main():
    """Import wallet and display its details."""
    print("initializing agentkit...")
    agentkit, config = initialize_agentkit()

    wallet_data = agentkit.export_wallet()

    print("ensuring wallet...")
    wallet = json.loads(wallet_data)

    print("wallet Id:", wallet["wallet_id"])
    print("wallet address:", wallet["default_address_id"])
    print("wallet network:", wallet["network_id"])
    print("wallet seed:", wallet["seed"])


if __name__ == "__main__":
    try:
        main()
    except Exception as error:
        print("Fatal error:", str(error))
        sys.exit(1)

File: cdp-langchain/examples/wallet/wallet.ts
import { CdpAgentkit } from "@coinbase/cdp-agentkit-core";
import { CdpToolkit } from "@coinbase/cdp-langchain";
import { HumanMessage } from "@langchain/core/messages";
import { MemorySaver } from "@langchain/langgraph";
import { createReactAgent } from "@langchain/langgraph/prebuilt";
import { ChatOpenAI } from "@langchain/openai";
import * as dotenv from "dotenv";
import * as fs from "fs";
import * as readline from "readline";

dotenv.config();

/**
 * Initialize the CDP Agentkit
 *
 * @returns Agent executor and config
 */
async function initializeAgentkit() {
  const config = {};
  const agentkit = await CdpAgentkit.configureWithWallet(config);

  return { agentkit, config };
}

/**
 * Import wallet
 */
async function main() {
  console.log("initializing agent...");
  const { agentkit, config } = await initializeAgentkit();

  const walletData = await agentkit.exportWallet();

  console.log("ensuring wallet...");
  const wallet = JSON.parse(walletData);

  console.log("wallet Id:", wallet.walletId);
  console.log("wallet address:", wallet.defaultAddressId);
  console.log("wallet network:", wallet.networkId);
  console.log("wallet seed:", wallet.seed);
}

if (require.main === module) {
  main().catch(error => {
    console.error("Fatal error:", error);
    process.exit(1);
  });
}
